test/compiler.test.js

```diff
@@ -1588,11 +1603,23 @@ describe('compiler', function() {
         sourceFileForUrlNotFound);
     });
 
+    it('Suports `middlewareRootDir` Option And ' +
+      'emits middleware instructions', function() {
+      testMiddlewareRegistration('loopback/server/middleware/url-not-found',
+        sourceFileForUrlNotFound, true);
```

Comment:
I find it rather difficult to understand what's happening under the hood and how is this different from the test above, because the single `true` argument does not convey almost any information.

I believe one test is enough to verify that middleware registration looks at the correct json file, there is no need to test all different flavours of middleware name, thus `testMiddlewareRegistration` can remain unmodified.

```js
it('supports `middlewareRootDir` option', function() {
  var middlewareJson = {
    custom: {
      'loopback#urlNotFound': {}
    }
  };
  var customDir = path.resolve(appdir.PATH, 'custom');
  fs.mkdirsSync(customDir);
  fs.writeJsonSync(path.resolve(customDir, 'middleware.json'), middlewareJson);

  var instructions = boot.compile({
    appRootDir: appdir.PATH,
    middlewareRootDir: path.resolve(appdir.PATH, 'custom')
  });
 
  expect(instructions.middleware).to.eql([{
    sourceFile: sourceFileForUrlNotFound,
    config: { phase: 'custom' }
  }]);
);
```